homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,147 @@
+"""
+Support for Spider thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.spider/
+"""
+
+import logging
+
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE, STATE_COOL, STATE_HEAT, STATE_IDLE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_OPERATION_MODE, ClimateDevice)
+from homeassistant.components.spider import DOMAIN as SPIDER_DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+
+DEPENDENCIES = ['spider']
+
+OPERATION_LIST = [
+    STATE_HEAT,
+    STATE_COOL,
+]
+
+HA_STATE_TO_SPIDER = {
+    STATE_COOL: 'Cool',
+    STATE_HEAT: 'Heat',
+}
+
+SPIDER_STATE_TO_HA = {value: key for key, value in HA_STATE_TO_SPIDER.items()}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Spider thermostat."""
+    thermostats = hass.data[SPIDER_DOMAIN]['thermostats']
```

Comment:
Add a check if discovery_info is None and return if so. We only want to set up the platform if it's the component that loaded the platform. Ie not via platform config.